app/utils/graphql.ts

```diff
@@ -0,0 +1,194 @@
+import { useState, useCallback, useEffect, useMemo } from "react";
+import useSWR, { SWRConfiguration } from "swr";
```

Comment:
all of our data fetching is managed with [SWR](https://swr.vercel.app/docs/getting-started) and [graphql-request](https://github.com/graffle-js/graffle/tree/graphql-request).

SWR is a Vercel-managed library designed specifically for client-side data fetching, and I really like working with it. 

The Apollo [setup for nextjs app router](https://www.apollographql.com/blog/using-apollo-client-with-next-js-13-releasing-an-official-library-to-support-the-app-router) looks fairly involved, and i haven't messed with it. I'm open to it, but I stuck with SWR since it's familiar to me and is part of the nextjs ecosystem.

Reply:
i think swr is great, and I don't see anything it doesn't cover that we got out of Apollo. imo Apollo client probably gives more than we need, and we also ran into some weird problems with fetching data for csv exports. Actually, lazy querying is something that we'll probably need, but maybe it is covered?